x/stakeibc/keeper/community_pool_test.go

```diff
@@ -0,0 +1,146 @@
+package keeper_test
+
+import (
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+
+	epochtypes "github.com/Stride-Labs/stride/v14/x/epochs/types"
+	icqtypes "github.com/Stride-Labs/stride/v14/x/interchainquery/types"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/keeper"
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+func (s *KeeperTestSuite) SetupQueryCommunityPoolBalance(icaAccountType types.ICAAccountType) types.HostZone {
+	// We need to register the transfer channel to initialize the light client state
+	s.CreateTransferChannel(HostChainId)
+
+	// Craete host zone
+	// We must use valid addresses for each ICA since they're serialized for the query request
+	depositAddress := s.TestAccs[0]
+	returnAddress := s.TestAccs[1]
+	hostZone := types.HostZone{
+		ChainId:                        HostChainId,
+		ConnectionId:                   ibctesting.FirstConnectionID,
+		CommunityPoolDepositIcaAddress: depositAddress.String(),
+		CommunityPoolReturnIcaAddress:  returnAddress.String(),
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	// Create epoch tracker for timeout
+	epochEndTime := uint64(1000)
+	epochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		NextEpochStartTime: epochEndTime,
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx, epochTracker)
+
+	return hostZone
+}
+
+// Helper function to verify the query that was submitted from the community pool balance query
+func (s *KeeperTestSuite) checkCommunityPoolQuerySubmission(
+	hostZone types.HostZone,
+	icaAccountType types.ICAAccountType,
+	expectedIcaAddress string,
```

Comment:
The hostZone and icaAccountType together should give enough info to compute the expectedIcaAddress so why include it as a 3rd input?  This would avoid the weird pattern of handing in 'Atom' as a filler value in the failure cases below?

Reply:
i went back and forth on this myself. My end decision was that it made more sense to hard code the expected address when invoking the test, rather than redefine the switch logic that's in the actual code in this function. 

this was the test reads a bit more explicitly and we don't run the risk of reintroducing a bug into our test (if there was a bug in the og function)

Reply:
ok... but this introduces a new case to test (where the icaAddress provided doesn't match the type given) which is impossible since the logic in the function computes the address.  I don't think I agree with this philosophy yet.  Seems cleaner to avoid it completely and also avoid the test cases where you are passing filler values

Reply:
hmm yeah that's a good point, re: address not matching type. I'll remove the param

and re: filler values, I think it's unavoidable - even if we remove the address string, the accountType is still a filler

Reply:
updated